Make `/coin check` default to the caller

hydrobeam opened the pull request:
# Related Issues
<!-- Issue(s) that this PR will resolve, e.g. "resolves <issue link here>". -->
N/A

# Summary of Changes
<!-- A summary or description of changes made in this PR. -->
In comparison to `.coin`, doing `/coin check` and having to look for the target user was pretty time consuming when somebody just wanted to check their own balance. This PR makes the `user` argument optional and defaults to checking the balance of the user who called the command if the arg is not provided. 

- Balance not present + coin check optional
<img width="706" alt="image" src="https://user-images.githubusercontent.com/70682032/184524478-8c032d9c-cd1b-418e-b23c-e5c6edb16fb9.png">

- .coin default behaviour checks out + message changes depending on how user is called
<img width="262" alt="image" src="https://user-images.githubusercontent.com/70682032/184525009-7a991475-44d1-4b82-9af9-b653af1f98c0.png">

# Steps to Reproduce
<!-- List the steps needed for the reviewer to produce your PR changes.
     If possible, also include screenshots to illustrate your new feature or fix. -->
`/coin check` and pass in no arguments.
OR
`.coin check` and pass in no arguments.

## Extra commentary

Not sure if `messageFromUser.member?.user` is the "right" way to get the user, since for some reason accessing `member` can return null from a `Message`?🤷

mcpenguin commented on src/commandDetails/coin/check.ts:
```diff
@@ -13,12 +13,11 @@ import { getCoinEmoji } from '../../components/emojis';
-  const user = <User>args['user'];
-
+  // use the caller as a default user if no argument is provided
+  const user = <User>args['user'] ?? messageFromUser.member?.user;
```
Does `messageFromUser.user` work?

hydrobeam replied:
Nope, `user` is not a valid field for a `Message`. `author` exists for message, but I don't really know in what situation accessing member could be null for a message 🤷 so it's probably not a concern

mcpenguin commented:
Also please leave screenshots :))

hydrobeam commented:
Decided to remove the `balance` subcommand since having `/coin check` default to self makes it redundant. To maintain appearances, I added a check to see if the caller is running the method without arguments and changes the initial message to `You have` accordingly.

Also reworked `getUserIDFromMessage` to be more general

mcpenguin commented on src/commandDetails/coin/check.ts:
```diff
@@ -5,28 +5,39 @@ import {
+  return `${displayMessage} ${balance} Codey coins ${getCoinEmoji()}.`;
 };
 
 export const coinCheckCommandDetails: CodeyCommandDetails = {
```
If you're removing balance, make `aliases` also have `balance, bal and b`, since people use these commands regularly.

hydrobeam replied:
Done

mcpenguin reviewed (changes requested):
small changes

mcpenguin reviewed (approved):
LGTM! Feel free to merge after you fix the linter.

hydrobeam commented:
> LGTM! Feel free to merge after you fix the linter.

Fixed the linter, but don't have write access :see_no_evil:
